synapse/handlers/oidc.py

```diff
@@ -684,9 +681,6 @@ async def _parse_id_token(self, token: Token, nonce: str) -> UserInfo:
             # If we got an `access_token`, there should be an `at_hash` claim
             # in the `id_token` that we can check against.
             claims_params["access_token"] = token["access_token"]
-            claims_cls = CodeIDToken
-        else:
-            claims_cls = ImplicitIDToken
```

Comment:
Looking again at what `CodeIDToken` and `ImplicitIDToken` verify in `authlib`, it really did not make sense to have that logic here.

Reply:
The only things it changes is the verification done on claims in the ID token. `CodeIDToken` is for the flow we're using, `ImplicitIDToken` is for the implicit flow (where you get the access token/ID token directly in the callback params).